lib/BuildSystem/BuildSystem.cpp

```diff
@@ -2074,7 +2075,24 @@ class ShellCommand : public ExternalCommand {
       virtual void actOnRuleDependency(const char* dependency,
                                        uint64_t length,
                                        const StringRef unescapedWord) override {
-        bsci.taskDiscoveredDependency(task, BuildKey::makeNode(unescapedWord));
+        if (llvm::sys::path::is_absolute(unescapedWord)) {
+          bsci.taskDiscoveredDependency(task, BuildKey::makeNode(unescapedWord));
+          return;
+        }
+
+        // Generate absolute path
+        //
+        // NOTE: This is making the assumption that relative paths coming in a
+        // dependency file are in relation to the explictly set working
+        // directory, or the current working directory when it has not been set.
+        SmallString<PATH_MAX> absPath = StringRef(command->workingDirectory);
+        if (!absPath.empty()) {
+          absPath.append("/");
+        }
+        absPath.append(unescapedWord);
+        llvm::sys::fs::make_absolute(absPath);
```

Comment:
Doesn't llvm::sys::fs have support for appending a path, versus the manual code above?

Reply:
It does indeed.  I looked for it before, but I was looking for "join" rather than "append" and overlooked it.